src/dbnode/persist/fs/commitlog/writer.go

```diff
@@ -224,22 +237,30 @@ func (w *writer) Write(
 		metadata.Namespace = series.Namespace.Bytes()
 		metadata.Shard = series.Shard
 		metadata.EncodedTags = encodedTags
-		w.metadataEncoder.Reset()
-		if err := w.metadataEncoder.EncodeLogMetadata(metadata); err != nil {
+
+		w.metadataEncoderBuff = w.metadataEncoderBuff[:0]
+		var err error
+
+		w.metadataEncoderBuff, err = msgpack.EncodeLogMetadataFast(w.metadataEncoderBuff, metadata)
```

Comment:
p.s. I realized this can be collapsed somewhat to do the `[:0]` just as you pass it to the EncodeLogMetadataFast method:
```go
var err error
w.metadataEncoderBuff, err = msgpack.EncodeLogMetadataFast(w.metadataEncoderBuff[:0], metadata)
```